sdk/core/core-lro/src/lroPoller/locationPolling.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { LROStatus, RawResponse } from "./models";
+
+function isLocationPollingDone(rawResponse: RawResponse): boolean {
+  const code = rawResponse.statusCode;
+  if (![202, 200].includes(code)) {
+    throw new Error(`Operation failed`);
```

Comment:
If this bubbles up, we should think about how to message it better, like `Received unexpected HTTP status code ${code} while polling. This may indicate a server issue.`

We should also figure out how we recommend devs recover from such a state, same with other exceptions we through in these internals

Reply:
Done. Not sure about the how to recover from it part though, are you saying we should customize the message based on the http code?

Reply:
For the first cut I don't think we need to go too crazy. To start, just think about what you'd want to see in a GitHub issue when a consumer is hitting a bug here because the server is misbehaving.

Reply:
Done.